gr-fec/lib/decode_ccsds_27_fb_impl.cc

```diff
@@ -66,6 +66,14 @@ int decode_ccsds_27_fb_impl::work(int noutput_items,
                 viterbi_get_output(d_state0, out++);
                 // printf("%li\n", *(out-1), metric);
             }
+
+            // Periodically normalize the path metrics to prevent overflow
+            if (d_count % 65536 == 65535) {
```

Comment:
With this interval, the metrics reach approximately 2^24 before being normalized. This keeps them well below overflow, which would occur at 2^31.